src/test/functional/functional-tests.js

```diff
@@ -1368,4 +1378,75 @@ describe('functional tests', function() {
 
     })
   })
+  describe('Versioning tests on a buckets: getObject, getPartialObject, putObject, removeObject with versionId support', function () {
+    //Isolate the bucket/object for easy debugging and tracking.
+    const  versionedBucketName = "minio-js-test-version-" + uuid.v4()
+    const versioned_100kbObjectName = 'datafile-versioned-100-kB'
+    const versioned_100kb_Object = dataDir ? fs.readFileSync(dataDir + '/' + versioned_100kbObjectName) : Buffer.alloc(100 * 1024, 0)
+
+    before((done) => client.makeBucket(versionedBucketName, '', done))
+    after((done) => client.removeBucket(versionedBucketName, done))
+
+    describe('Versioning Test for  getObject, getPartialObject, putObject, removeObject with versionId support', function () {
+      let versionId=null
+      step("Enable Versioning on Bucket ",(done)=>{
+        client.setBucketVersioning(versionedBucketName,{Status:"Enabled"},(err)=>{
+          if (err && err.code === 'NotImplemented') return done()
+          if (err) return done(err)
+          done()
+        })
+      })
+
+      step(`putObject(bucketName, objectName, stream)_bucketName:${bucketName}, objectName:${versioned_100kbObjectName}, stream:100Kib_`, done => {
+        client.putObject(versionedBucketName, versioned_100kbObjectName, versioned_100kb_Object)
+          .then((res={}) => {
+            if(res.versionId){
+              versionId = res.versionId
+              done()
+            }else{
+              done(new Error('versionId not found in putObject response'))
+            }
+          })
+          .catch(done)
+      })
+
+      step(`getObject(bucketName, objectName, getOpts) :${versionedBucketName}, objectName:${versioned_100kbObjectName}`, done => {
```

Comment:
I thought I gave this comment, but I cannot see it.

In all these `step` explanations, the standard is to have `_bucketName` before bucket name.

```suggestion
      step(`getObject(bucketName, objectName, getOpts)_bucketName :${versionedBucketName}, objectName:${versioned_100kbObjectName}`, done => {
```

There are several cases in your PR.

Reply:
Thank you. I have applied the suggestion.